### DISCUSSION TITLE
Clearance needed

### DISCUSSION THREAD

**Discussion by sassanh** [ant-design/ant-design#33950] [General]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
AntD is a great design system that works very good out of the box, good job on that!
There are minor issues here and there and it's totally expected for such a big library.

But AntD should make it clear for its users: Is it a library which is trying to be an integrated solution **intending** to work perfectly out of the box? Or is it a library that provides most of the good stuff and if there is a minor bug, the developer team may close the bug report and ask the user to apply a hack to workaround the issue?

Note that the decision is not to be perfect or imperfect, ofc we can't avoid minor issues, it is just the intention that matters, are we **trying** to move toward being perfect and acknowledge problems so that we can eventually resolve them?

There are many reported issues that have been addressed perfectly by the community by providing quality solutions or quality fix in the library, etc. But here https://github.com/ant-design/ant-design/issues/14731 and here https://github.com/ant-design/ant-design/issues/15744 it seems like developer team is preferring to prioritize closing issues to have less open issues in issue list by providing hacky solutions and assuming these hacky solutions are enough to address the reported problem rather than keeping the issue open, acknowledging there is a problem and trying to eventually resolve it.

It is completely understandable for an open source library to have long lasting open issues that no one is interested to work on, I personally consider such library that acknowledges problems and keep track of them a high quality library even if these problems are not resolved for many years compared to a library that suggests hacks and closes unresolved open issues.

Looking forward to read others opinion.

### What does the proposed API look like?
It is more of a meta issue, there is no proposed API.

**Comment by sassanh** [comment]
Thanks for converting it to a discussion!
We found another similar issue in which a developer team member is suggesting a hack and closing the issue like the hack is enough to address the problem: https://github.com/ant-design/ant-design/issues/25048

**Comment by afc163** [comment]
Sorry for delay reply, for https://github.com/ant-design/ant-design/issues/25048 I think we can add new prop `theme` to `<Header />`. Feel free to open new issue about feature request.

**Comment by afc163** [comment]
Looking forword to better idea for new feature request rather than hacky workaround in old issues.

**Comment by sassanh** [comment]
I'm glad you are positive about this.

A helpful action would be to mention it in the `README.md` that we are willing to acknowledge and hopefully fix problems rather than sweeping them under the carpet. By writing this, hopefully it will become the sprint between developers and the users and avoid closing valid issues in the future. I think it will bring good reputation for AntD.
